go/vt/sqlparser/sql_test.go

```diff
@@ -280,6 +280,264 @@ func TestShowIndex(t *testing.T) {
+				id INT,
+				parent_id INT,
+				INDEX par_ind (parent_id),
+				FOREIGN KEY (parent_id)
+					REFERENCES parent(id)
+					ON UPDATE RESTRICT ON DELETE CASCADE
+			)`,
+			[]*ConstraintDefinition{
+				{
+					Name: "",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parent_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parent"}},
+						ReferencedColumns: Columns{{val: "id"}},
+						OnUpdate: Restrict,
+						OnDelete: Cascade,
+					},
+				},
+			},
+		},
+		{
+			`CREATE TABLE child (
+				id INT,
+				parent_id INT,
+				INDEX par_ind (parent_id),
+				FOREIGN KEY (parent_id)
+					REFERENCES parent(id)
+					ON DELETE NO ACTION ON UPDATE SET DEFAULT
+			)`,
+			[]*ConstraintDefinition{
+				{
+					Name: "",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parent_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parent"}},
+						ReferencedColumns: Columns{{val: "id"}},
+						OnUpdate: SetDefault,
+						OnDelete: NoAction,
+					},
+				},
+			},
+		},
+		{
+			`CREATE TABLE child (
+				id INT,
+				parent_id INT,
+				INDEX par_ind (parent_id),
+				FOREIGN KEY (parent_id)
+					REFERENCES parent(id)
+					ON DELETE CASCADE
+			)`,
+			[]*ConstraintDefinition{
+				{
+					Name: "",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parent_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parent"}},
+						ReferencedColumns: Columns{{val: "id"}},
+						OnUpdate: DefaultAction,
+						OnDelete: Cascade,
+					},
+				},
+			},
+		},
+		{
+			`CREATE TABLE child (
+				id INT,
+				parent_id INT,
+				INDEX par_ind (parent_id),
+				FOREIGN KEY (parent_id)
+					REFERENCES parent(id)
+					ON UPDATE SET NULL
+			)`,
+			[]*ConstraintDefinition{
+				{
+					Name: "",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parent_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parent"}},
+						ReferencedColumns: Columns{{val: "id"}},
+						OnUpdate: SetNull,
+						OnDelete: DefaultAction,
+					},
+				},
+			},
+		},
+		{
+			`CREATE TABLE child (
+				id INT,
+				parents_id INT,
+				parents_sub_id INT,
+				INDEX par_ind (parents_id, parents_sub_id),
+				CONSTRAINT fk_parents FOREIGN KEY (parents_id, parents_sub_id)
+					REFERENCES parents(id, sub_id)
+			)`,
+			[]*ConstraintDefinition{
+				{
+					Name: "fk_parents",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parents_id"}, {val: "parents_sub_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parents"}},
+						ReferencedColumns: Columns{{val: "id"}, {val: "sub_id"}},
+						OnUpdate: DefaultAction,
+						OnDelete: DefaultAction,
+					},
+				},
+			},
+		},
+	}
+
+	testForeignKey(t, tests)
+}
+
+func TestAlterAddForeignKey(t *testing.T) {
+	tests := []testForeignKeyStruct{
+		{
+			`ALTER TABLE child ADD
```

Comment:
Same comment about testing more syntax permutations. Need one that does ADD FOREIGN KEY (cols) REFERNCES ...  without any ON UPDATE or ON DELETE